### PR TITLE
Feature/js19 431 angular authorization

### PR DESCRIPTION
**Pull request by EldrAbdr**

### CODE REVIEW

File: apps/angular/src/app/app.module.ts [hunk 1]
```diff
@@ -11,15 +11,23 @@ import { MatSortModule } from '@angular/material/sort';
 import { MatPaginatorModule } from '@angular/material/paginator';
 import { MatFormFieldModule } from '@angular/material/form-field';
 import { MatSelectModule } from '@angular/material/select';
-import { ReactiveFormsModule } from '@angular/forms';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+
+import { MatButtonModule } from '@angular/material/button';
+
+import { MatToolbarModule } from '@angular/material/toolbar';
```
Comment (panov-aleksey-saritasa) [inline apps/angular/src/app/app.module.ts:18]: Please, group imports of the same level together

File: apps/angular/src/app/features/anime/components/anime-table/anime-table.component.css [hunk 2]
```diff
@@ -1,6 +1,13 @@
 .anime-page {
   background-color: var(--primary-color);
   width: 100%;
+  color: var(--primary-contrast-color);
+}
+
+.header {
+  display: flex;
+  justify-content: space-between;
+  margin-bottom: 70px;
```
Comment (panov-aleksey-saritasa) [inline apps/angular/src/app/features/anime/components/anime-table/anime-table.component.css:10]: According to the [BEM docs](https://en.bem.info/methodology/quick-start/#block):

> The block shouldn't influence its environment, meaning you shouldn't set the external geometry (margin) or positioning for the block.

File: apps/angular/src/app/features/anime/components/anime-table/anime-table.component.html [hunk 3]
```diff
@@ -1,22 +1,32 @@
 <div class="anime-page">
+  <div class="header">
```
Comment (panov-aleksey-saritasa) [inline apps/angular/src/app/features/anime/components/anime-table/anime-table.component.html:2]: To be more semantically correct, use [header tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/header) instead

File: apps/angular/src/app/features/anime/components/anime-table/anime-table.component.html [hunk 4]
```diff
@@ -1,22 +1,32 @@
...
+    <h1 class="header__title">Anime catalog</h1>
+    <a
+      mat-stroked-button
+      (click)="handleQuitButton()"
+      routerLink="/login"
+      class="button"
+      type="button"
+    >
+      Quit
+    </a>
```
Comment (panov-aleksey-saritasa) [inline apps/angular/src/app/features/anime/components/anime-table/anime-table.component.html:12]: Usually we use `<a>` tag for navigation purposes and `<button>` when we need to do some logic on click, consider not mixing these things up

File: apps/angular/src/app/features/anime/components/anime-table/anime-table.component.ts [hunk 5]
```diff
@@ -295,4 +297,9 @@ export class AnimeTableComponent implements OnInit, AfterViewInit, OnDestroy {
   public ngOnDestroy(): void {
     this.subscriptionOnChanges.unsubscribe();
   }
+
+  /** Handle quit button click. */
+  public handleQuitButton(): void {
+    this.userService.removeJwtFromLocalStorage();
+  }
```
Comment (panov-aleksey-saritasa) [inline apps/angular/src/app/features/anime/components/anime-table/anime-table.component.ts:304]: On Angular projects we use `on` prefix for event handlers, also your method name is missing high context ([wiki](https://wiki.saritasa.rocks/frontend/languages/ts-js/naming#ahclc-pattern))
```suggestion
  /** Handle quit button click. */
  public onQuitButtonClick(): void {
    this.userService.removeJwtFromLocalStorage();
  }
```

File: apps/angular/src/app/features/auth/login/login.component.ts [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+  /** Error message received from the server. */
+  public responseErrorMessage: string;
+
+  public constructor(
+    private readonly formBuilder: FormBuilder,
+    private readonly auth: AuthorizationService,
+    private readonly router: Router,
+    private readonly userService: UserService,
+    private change: ChangeDetectorRef,
```
Comment (panov-aleksey-saritasa) [inline apps/angular/src/app/features/auth/login/login.component.ts:41]:
```suggestion
    private readonly change: ChangeDetectorRef,
```

File: apps/angular/src/app/features/auth/login/login.component.ts [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    this.form = this.formBuilder.group({
+      email: ['', [Validators.required, Validators.email]],
+      password: ['', Validators.required],
+    });
+    this.responseErrorMessage = '';
+  }
+
+  /** Handle login form submit. */
+  public submitLogin(): void {
+    this.auth.login(this.form.value).subscribe({
```
Comment (panov-aleksey-saritasa) [inline apps/angular/src/app/features/auth/login/login.component.ts:52]: Don't forget to unsubscribe https://wiki.saritasa.rocks/frontend/frameworks/angular/best-practices#manage-your-subscriptions

File: apps/angular/src/index.html [hunk 8]
```diff
@@ -10,7 +10,7 @@
     <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@300;400;500&display=swap" rel="stylesheet">
     <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
 </head>
-  <body>
+  <body style="background-color: var(--primary-color)">
```
Comment (panov-aleksey-saritasa) [inline apps/angular/src/index.html:13]: you could use either `libs/theme/src/index.css` or `apps/angular/src/theme/styles.css` for this

File: apps/angular/src/app/features/auth/login/login.component.ts [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+    });
+  }
+
+  /**
+   * Handler of success response after registration.
+   * @param userToken Authorization token.
+   */
+  private handleSuccessResponse(userToken: UserToken): void {
+    this.router.navigate(['/anime']);
+    this.userService.saveJwtInLocalStorage(userToken.jwt);
```
Comment (panov-aleksey-saritasa) [inline apps/angular/src/app/features/auth/login/login.component.ts:64]: Consider doing this in `login` method instead, I think your UI components shouldn't know anything about tokens/jwt and etc.

File: apps/angular/src/app/app.module.ts (hunk 1 above)
Comment (EldrAbdr) [reply apps/angular/src/app/app.module.ts]: Done

File: apps/angular/src/app/features/anime/components/anime-table/anime-table.component.css (hunk 2 above)
Comment (EldrAbdr) [reply apps/angular/src/app/features/anime/components/anime-table/anime-table.component.css]: Corrected

File: apps/angular/src/app/features/anime/components/anime-table/anime-table.component.html (hunk 3 above)
Comment (EldrAbdr) [reply apps/angular/src/app/features/anime/components/anime-table/anime-table.component.html]: Corrected

File: apps/angular/src/app/features/anime/components/anime-table/anime-table.component.html (hunk 4 above)
Comment (EldrAbdr) [reply apps/angular/src/app/features/anime/components/anime-table/anime-table.component.html]: Replaced with button tag

File: apps/angular/src/app/features/anime/components/anime-table/anime-table.component.ts (hunk 5 above)
Comment (EldrAbdr) [reply apps/angular/src/app/features/anime/components/anime-table/anime-table.component.ts]: Renamed

File: apps/angular/src/app/features/auth/login/login.component.ts (hunk 7 above)
Comment (EldrAbdr) [reply apps/angular/src/app/features/auth/login/login.component.ts]: Done

File: apps/angular/src/app/features/auth/login/login.component.ts (hunk 6 above)
Comment (EldrAbdr) [reply apps/angular/src/app/features/auth/login/login.component.ts]: Corrected

File: apps/angular/src/index.html (hunk 8 above)
Comment (EldrAbdr) [reply apps/angular/src/index.html]: Done

File: apps/angular/src/app/features/auth/login/login.component.ts (hunk 9 above)
Comment (EldrAbdr) [reply apps/angular/src/app/features/auth/login/login.component.ts]: Actions with response moved to auth service

**Pull request closed without merging**